api/test_views.py

```diff
@@ -418,3 +418,151 @@ def test_admin2_api(self):
+        admin2_2.is_deprecated = False
+        admin2_1.save()
+        admin2_2.save()
+        response = self.client.get('/api/v2/admin2/').json()
+        self.assertEqual(response['count'], 3)
+
+        admin2_1.is_deprecated = True
+        admin2_2.is_deprecated = True
+        admin2_3.is_deprecated = True
+        admin2_1.save()
+        admin2_2.save()
+        admin2_3.save()
+        # All admin2-s are deprecated
+        response = self.client.get('/api/v2/admin2/').json()
+        self.assertEqual(response['count'], 0)
+
+        # Tear down
+        admin2_1.is_deprecated = False
+        admin2_2.is_deprecated = False
+        admin2_3.is_deprecated = False
+        admin2_1.save()
+        admin2_2.save()
+        admin2_3.save()
+        response = self.client.get('/api/v2/admin2/').json()
+        self.assertEqual(response['count'], 3)
+
+        admin1_2.is_deprecated = True
+        admin1_2.save()
+        response = self.client.get('/api/v2/admin2/').json()
+        # There are 2 admin2-s in this district, so 3-2 = 1
+        self.assertEqual(response['count'], 1)
+
+        # Tear down
+        admin1_2.is_deprecated = False
+        admin1_2.save()
+        response = self.client.get('/api/v2/admin2/').json()
+        self.assertEqual(response['count'], 3)
+
+        admin1_1.is_deprecated = True
+        admin1_1.save()
+        response = self.client.get('/api/v2/admin2/').json()
+        # There is only 1 admin2-s in this district, so 3-1 = 2
+        self.assertEqual(response['count'], 2)
+
+        # Tear down
+        admin1_1.is_deprecated = False
+        admin1_1.save()
+        response = self.client.get('/api/v2/admin2/').json()
+        self.assertEqual(response['count'], 3)
+
+        country.is_deprecated = True
+        country.save()
+        response = self.client.get('/api/v2/admin2/').json()
+        # There are 3 admin2-s in this country, so 3-3 = 0
+        self.assertEqual(response['count'], 0)
+
+        # Tear down
+        country.is_deprecated = False
+        country.save()
+        response = self.client.get('/api/v2/admin2/').json()
+        self.assertEqual(response['count'], 3)
+
+
+class DistrictTest(APITestCase):
+    def test_district_deprecation(self):
+        region = models.Region.objects.create(name=1)
+        country = models.Country.objects.create(name='Nepal', iso3='NPL', region=region)
+        admin1_1 = models.District.objects.create(name='admin1 1', code='NPL01', country=country)
+        admin1_2 = models.District.objects.create(name='admin1 2', code='NPL02', country=country)
+
+        # test fetching all districts
+        response = self.client.get('/api/v2/district/').json()
+        self.assertEqual(response['count'], 2)
+
+        admin1_2.is_deprecated = True
+        admin1_2.save()
+        response = self.client.get('/api/v2/district/').json()
+        # one district deprecated, 2-1 = 1
+        self.assertEqual(response['count'], 1)
+
+        # Tear down
```

Comment:
Am not sure if this `Tear down` comment makes sense in these places? Is that intentional?

Reply:
Yes. My intention was to show: with that code part I "make back" the disturbed situation to a "nothing is deprecated" situation. Shall I use other expression for that?

Reply:
Ah I see, makes sense. That's fine - as long as it was intentional, that is fine.